plugin/codeclimate.vim

```diff
@@ -30,17 +30,21 @@ function! s:AnalyzeOpenFiles()
     call s:ModifiedFilesWarning()
   endif
 
+  let l:extraopts = s:BufferAnalysisOpts(l:bufs)
   let l:files = s:BufferNames(l:bufs)
+  let l:cli = join(l:extraopts, ' ') . ' ' . join(map(l:files, 'shellescape(v:val)'), ' ')
 
-  call s:RunAnalysis(join(map(l:files, 'shellescape(v:val)'), ' '))
+  call s:RunAnalysis(l:cli)
 endfunction
 
 function! s:AnalyzeCurrentFile()
   if &modified
     call s:ModifiedFilesWarning()
   endif
 
-  call s:RunAnalysis(shellescape(@%))
+  let l:cli = getbufvar(@%, 'codeclimateflags') . ' ' . shellescape(@%)
```

Comment:
It gets a bit verbose, and is doing some rather pointless processing:

```
let l:cli = join(s:BufferAnalysisOpts([@%]), ' ') . ' ' . shellescape(@%)
```

If your concern is the "magic string" leakage, I was annoyed by that too. Extracting the var name to a script-scoped var & using that instead of the literal string would eliminate that.